test/lint/lint-assertions.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
```

Comment:
The `lint-assertions.sh` script needs to be added to `lint-all.sh` as we don't pick up lint scripts automatically as upstream does, but call them explicitly by their name.

The same is true for other scripts which have been added by the upstream sync.